### PR TITLE
Issue 1370: Renaming Pravega Metrics

### LINKED ISSUE #1370: Renaming Metrics
**Problem description**
The naming of the metrics are a little disorganized.  Also, we want to come up with a better way to name metrics so its easier to filter the metrics.

**Problem location**
Metrics

**Suggestions for an improvement**
1.  Prefix all metrics with "pravega".
2.  Remove "DYNAMIC" from metric names as its not required by anyone consuming the metrics.
3.  Adopt the following format for naming:

pravega.<source>.<metric_name>.<tags...>

So for example, metrics would look like:

pravega.store.segment_read_bytes.$scope.$stream.$segment
pravega.controller.segment_splits.$scope.$stream
pravega.controller.tier2_total_used

These changes would allow us to filter all pravega metrics to be stored in its own database.  Also, we can write templates that know the number of tags for each metric name.

### PR DESCRIPTION
**Pull request by Abhijeet-Jadhav**
**Change log description**
Renaming metrics reported in pravega.

**Purpose of the change**
Fixes #1370 

**What the code does**
Renames metrics in pravega. Removes 'DYNAMIC' from names. Adds relevant prefix controller, store, hdfs etc.

**How to verify it**
startStandlaone gradle task will start pravega services and some metrics can be seen in /tmp/csv/pravega folder.

### CODE REVIEW

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java [hunk 1]
```diff
@@ -100,7 +100,7 @@
     static final Duration TIMEOUT = Duration.ofMinutes(1);
     static final int MAX_READ_SIZE = 2 * 1024 * 1024;
 
-    private static final StatsLogger STATS_LOGGER = MetricsProvider.createStatsLogger("host");
+    private static final StatsLogger STATS_LOGGER = MetricsProvider.createStatsLogger("store");
```
Comment (bayarsaikhan) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java]: store is bit confusing. I'd suggest "segmentstore"

File: shared/metrics/src/test/java/io/pravega/shared/metrics/MetricsProviderTest.java [hunk 2]
```diff
@@ -145,7 +146,8 @@ public void testContinuity() {
                                             .build();
         MetricsProvider.initialize(config);
 
-        Assert.assertNotNull(null, MetricsProvider.METRIC_REGISTRY.getCounters().get("continuity-counter"));
+        Assert.assertNotNull(null,
+                MetricsProvider.METRIC_REGISTRY.getCounters().get("pravega.testStatsLogger.continuity-counter"));
```
Comment (bayarsaikhan) [inline shared/metrics/src/test/java/io/pravega/shared/metrics/MetricsProviderTest.java]: What is this "testStatsLogger" in the middle of this one?

File: shared/metrics/src/test/java/io/pravega/shared/metrics/MetricsProviderTest.java [hunk 3]
```diff
@@ -107,8 +107,8 @@ public void testGauge() {
         for (int i = 1; i < 10; i++) {
             value.set(i);
             dynamicLogger.reportGaugeValue("dynamicGauge", i);
-            assertEquals(i, MetricsProvider.METRIC_REGISTRY.getGauges().get("testGauge").getValue());
-            assertEquals(i, MetricsProvider.METRIC_REGISTRY.getGauges().get("DYNAMIC.dynamicGauge.Gauge").getValue());
+            assertEquals(i, MetricsProvider.METRIC_REGISTRY.getGauges().get("pravega.testStatsLogger.testGauge").getValue());
+            assertEquals(i, MetricsProvider.METRIC_REGISTRY.getGauges().get("pravega.dynamicGauge.Gauge").getValue());
```
Comment (bayarsaikhan) [inline shared/metrics/src/test/java/io/pravega/shared/metrics/MetricsProviderTest.java]: Static metrics will also include 'pravega' prefix correct?

Comment (Abhijeet-Jadhav) [inline shared/metrics/src/test/java/io/pravega/shared/metrics/MetricsProviderTest.java]: Yes. See all names in issue comment: https://github.com/pravega/pravega/issues/1370#issuecomment-307305905

File: shared/metrics/src/test/java/io/pravega/shared/metrics/MetricsProviderTest.java (hunk 2 above)
Comment (Abhijeet-Jadhav) [inline shared/metrics/src/test/java/io/pravega/shared/metrics/MetricsProviderTest.java]: That is unit test for testing stats logger. Hence the name.

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java (hunk 1 above)
Comment (Abhijeet-Jadhav) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java]: Done.

**Pull request merged**